### ISSUE TITLE
Using setuptools generates egg-info folders in source space when installing

### ISSUE THREAD

**Issue by matthew-reynolds** [ros/catkin#1069]
Continuing the conversation from https://github.com/ros/catkin/pull/1048#discussion_r392361424. Using `setuptools`, as is now recommended:

`egg-info` folders are created in the source space when building Melodic packages w/ python 2 on Bionic. This seems to only happen when installing (ie. `catkin_make install`). (See https://travis-ci.org/github/ros-controls/ros_control/jobs/661982516?utm_medium=notification&utm_source=github_status)

Steps to reproduce:

```
cd /tmp
mkdir -p mel_ws/src
cd mel_ws/src
git clone git@github.com:ros/angles.git
chmod a-w -R angles
cd /tmp/mel_ws
catkin config --install; catkin build         # Fails
catkin clean
catkin_make install                           # Fails
catkin_make_isolated --install                # Fails
```

**Comment by matthew-reynolds** [comment]
FYI @sloretz @mikepurvis

**Comment by mikepurvis** [comment]
Oh, it's a Py2 issue—that might explain why I'm not hitting it as we're now migrated.

**Comment by dirk-thomas** [comment]
What `setuptools` version are you using (`pip freeze --all | grep setuptools`)?

**Comment by sloretz** [comment]
> What setuptools version are you using (pip freeze --all | grep setuptools)?

I'm able to reproduce with `setuptools==39.0.1`

**Comment by matthew-reynolds** [comment]
> What setuptools version are you using (pip freeze --all | grep setuptools)?

I'm using `setuptools==41.0.1`. Not sure what version the Travis build I linked used though.

Edit: I can still reproduce the issue with `setuptools=44.0.0`, which AFAIK is the latest version that is Py2.7 compatible.

**Comment by dirk-thomas** [comment]
Addressed in #1073.
